delphi-frontend/src/main/java/au/com/integradev/delphi/type/intrinsic/IntrinsicArgumentMatcher.java

```diff
@@ -20,12 +20,29 @@
 
 import au.com.integradev.delphi.type.TypeImpl;
 import au.com.integradev.delphi.type.TypeUtils;
+import org.sonar.plugins.communitydelphi.api.type.IntrinsicType;
 import org.sonar.plugins.communitydelphi.api.type.StructKind;
 import org.sonar.plugins.communitydelphi.api.type.Type;
 
 public final class IntrinsicArgumentMatcher extends TypeImpl {
-  public static final Type ANY_DYNAMIC_ARRAY =
-      new IntrinsicArgumentMatcher("<dynamic array>", Type::isDynamicArray);
+  public static final Type LIKE_DYNAMIC_ARRAY =
+      new IntrinsicArgumentMatcher(
+          "<like dynamic array>",
```

Comment:
Might be good to name this `<dynamic array, array constructor, string, or char>`, just because
`assertion failed: expected <like dynamic array>` is a bit of a confusing message.